### PR TITLE
BUG: Join behaved like an inner join when only one side had MultiIndex

### LINKED ISSUE #34292: BUG: join between df with a single index and df with a multiindex are always inner joins
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample

```python
A = pd.DataFrame([1,2], columns=["i"]).set_index(["i"])
B = pd.DataFrame([(1,4),(3,0),(1,5)], columns=["i", "ii"]).set_index(["i", "ii"])
A.join(B, how="left")
#Empty DataFrame
#Columns: []
#Index: [(1, 4), (1, 5)]
```

#### Problem description

Joining a df with 1 index and a df with a multiindex always generates an inner join, no matter the value of "how". In this case index-value 2 from the left df is missing, while it should be kept.

The same happens when just joining the indexes. This problem does not happen when joining 2 multi-indexes.

It seems something in the implementation in file pandas\core\indexes\base.py , class Index, method _join_level goes wrong. The generated new codes are either calculated wrong or don't take into account the join-method. But that's as far as I've gotten.

#### Expected Output

Empty DataFrame
Columns: []
Index: [(1, 4), (1, 5), (2, nan)]

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here leaving a blank line after the details tag]

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #34292
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

We can use the regular path now, since ``droplevel`` can handle an ``Index``. Only difference is a resorted result, because ``Index.join()`` does not preserve the order of the leading input

```
left = pd.Index([0, 1, 2, 1], name="b")
right = pd.Index([1, 2, 5], name="b")
print(left.join(right, how="left"))
```

returns 

```
Int64Index([0, 1, 1, 2], dtype='int64', name='b')
```

Edit: We may be able to delete the ``_join_level`` function, if we add a ``level`` keyword to the ``_join_multi`` function to determine the overlap.

### CODE REVIEW

**Comment by phofl** [comment]
The change caused a few other tests to fail, which I have not run locally. All because the new result has a differently sorted index. Is this ok or do we have to figure a way out to get the old behavior?

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -474,6 +474,7 @@ Reshaping
 - Bug in func :meth:`crosstab` when using multiple columns with ``margins=True`` and ``normalize=True`` (:issue:`35144`)
 - Bug in :meth:`DataFrame.agg` with ``func={'name':<FUNC>}`` incorrectly raising ``TypeError`` when ``DataFrame.columns==['Name']`` (:issue:`36212`)
 - Bug in :meth:`Series.transform` would give incorrect results or raise when the argument ``func`` was dictionary (:issue:`35811`)
+- Bug in :func:`join` behaved like an inner join, when one side had a regular :class:`Index` while the other side had a :class:`MultiIndex` (:issue:`34292`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:477]: move to 1.3

**Review by jreback — CHANGES_REQUESTED** [review]
@phofl if you can merge master will have a look

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (phofl) [reply doc/source/whatsnew/v1.2.0.rst]: Done

**Comment by jreback** [comment]
@phofl status here?

**Comment by simonjayhawkins** [comment]
@phofl closing as stale. reopen when ready.

**Pull request closed without merging**